packages/@aws-cdk/aws-iam/lib/oidc-provider/external.ts

```diff
@@ -8,7 +8,16 @@ import * as url from 'url';
 let client: aws.IAM;
 
 function iam() {
-  if (!client) { client = new aws.IAM(); }
+  if (!client) {
+    // China regions share same IAM endpoint, https://docs.amazonaws.cn/en_us/general/latest/gr/iam-service.html
+    if (process.env.AWS_REGION && process.env.AWS_REGION.startsWith('cn-')) {
```

Comment:
You shouldn't depend on `process.env` here, because the stack you're currently operating under might be using a different region.

Reply:
This OIDC provider is a custom resource of CloudFormation. Per [the doc of custom resource](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-custom-resources.html), I think the custom resource should be the same region of the resource be creating, it's OIDC provider here.

> Whenever anyone uses the template to create, update, or delete a custom resource, AWS CloudFormation sends a request to the specified service token. The service token must be in the same region in which you are creating the stack.

Also the IAM is a global service per partition, all regions in same partition share the same IAM service.

If the `process.env.AWS_REGION ` is not reliable sometimes, do you have any suggestion for  handling it more gracefully?